web/react/stores/socket_store.jsx

```diff
@@ -14,14 +12,25 @@ var CHANGE_EVENT = 'change';
 
 var conn;
 
-var SocketStore = assign({}, EventEmitter.prototype, {
-    initialize: function() {
+class SocketStoreClass extends EventEmitter {
+    constructor() {
+        super();
+
+        this.initialize = this.initialize.bind(this);
+        this.emitChange = this.emitChange.bind(this);
+        this.addChangeListener = this.addChangeListener.bind(this);
+        this.removeChangeListener = this.removeChangeListener.bind(this);
+        this.sendMessage = this.sendMessage.bind(this);
+
+        this.initialize();
+    }
+    initialize() {
         if (!UserStore.getCurrentId()) {
             return;
         }
 
         var self = this;
```

Comment:
We're supposed to be binding functions instead of saving references to this according to the style guide.